lib/games_collection.rb

```diff
@@ -14,4 +14,32 @@ def initialize(games_path)
   def create_games(games_path)
     create_instances(games_path, Game)
   end
+
+  def average_goals_per_game
+    goals_per_game = @games.map {|game| game.away_goals + game.home_goals}
+    all_goals = goals_per_game.sum
+    number_of_games = goals_per_game.length
+    average_goals_per_game = all_goals / number_of_games.to_f
+    average_goals_per_game.round(2)
+  end
+
+  def average_goals_by_season
+    season_to_game = games.reduce({}) do |acc, game|
```

Comment:
Might have an issue with keys being overridden. might need to employ a .has_key() method.